### ISSUE TITLE
RemovedUnusedBrs breaks validation on "select value may not be a tuple"

### ISSUE THREAD

**Issue by kripken** [WebAssembly/binaryen#3990]
Reduced fuzz testcase (10991288587553494659):
```wat
(module
 (type $none_=>_i32 (func (result i32)))
 (type $i64_externref_=>_v128 (func (param i64 externref) (result v128)))
 (type $none_=>_i32_externref_v128_funcref_eqref_f32 (func (result i32 externref v128 funcref eqref f32)))
 (type $none_=>_none (func))
 (memory $0 (shared 16 17))
 (func $0 (result i32)
  (unreachable)
 )
 (func $1 (param $0 i64) (param $1 externref) (result v128)
  (drop
   (block $label$4 (result i32 externref v128 funcref eqref f32)
    (drop
     (br_if $label$4
      (tuple.make
       (i32.const 67108865)
       (ref.null extern)
       (v128.const i32x4 0x00000000 0xc0100000 0x00000000 0x3ff00000)
       (ref.null func)
       (ref.null eq)
       (f32.load align=2
        (i32.const 4)
       )
      )
      (i32.const 0)
     )
    )
    (tuple.make
     (i32.const 0)
     (ref.null extern)
     (v128.const i32x4 0x00000000 0x00000000 0x00000000 0x00000000)
     (ref.null func)
     (ref.null eq)
     (f32.const 0)
    )
   )
  )
  (v128.const i32x4 0x00000000 0x00000000 0x00000000 0x00000000)
 )
 (func $2
  (nop)
 )
)
```
STR: `wasm-opt -all --remove-unused-brs`. That will emit a select on the two possible tuple values, which hits that error.

@tlively Do you remember why select values cannot be tuples?

**Comment by tlively** [comment]
The spec does not allow multivalue selects: https://webassembly.github.io/spec/core/valid/instructions.html#xref-syntax-instructions-syntax-instr-parametric-mathsf-select-t-ast.

**Comment by MaxGraey** [comment]
@tlively Btw does unity tuples (tuple with single element) lowered (optimize) to normal param and local value currently?

**Comment by tlively** [comment]
We currently disallow tuples with just one element. If you try to construct a tuple type with one element, you will get back the element type rather than a new type. `tuple.make` with just one argument is a validation error.

**Comment by MaxGraey** [comment]
Got it! Thanks

**Comment by kripken** [comment]
Thanks @tlively !

Fix in #4005
